Libraries/Opc.Ua.Gds.Server.Common/CertificateGroup.cs

```diff
@@ -304,6 +304,11 @@ string subjectName
             // initialize revocation list
             await RevokeCertificateAsync(AuthoritiesStorePath, newCertificate, null).ConfigureAwait(false);
 
+            if(SubjectName != subjectName)
```

Comment:
I have a few concerns to this change:
- throwing ArgumentException after some code already executes is a quite unusual pattern. 
- The certificate subject should be compared using the distinguished name class, as binary compare. 
Maybe this class needs some refactoring to ensure the issuer remain the same.

Reply:
I moved this line to the top of the class & use binare compare.

In the longterm the following decision should be made:
 - remove paramter "subjectName" -> this will change the signature of a public method & also the ICertificateGroupInterface
 - make it possible to change the CertificateGroup Configuration  in the Method by supplying the subjectName parameter